src/libraries/System.Diagnostics.Process/tests/ProcessTests.Windows.cs

```diff
@@ -15,5 +19,41 @@ private string WriteScriptFile(string directory, string name, int returnValue)
             File.WriteAllText(filename, $"exit {returnValue}");
             return filename;
         }
+        
+        private static void SendSignal(PosixSignal signal, int processId)
+        {
+            uint dwCtrlEvent = signal switch
+            {
+                PosixSignal.SIGINT => Interop.Kernel32.CTRL_C_EVENT,
+                PosixSignal.SIGQUIT => Interop.Kernel32.CTRL_BREAK_EVENT,
+                _ => throw new ArgumentOutOfRangeException(nameof(signal))
+            };
+
+            if (!Interop.GenerateConsoleCtrlEvent(dwCtrlEvent, (uint)processId))
+            {
+                int error = Marshal.GetLastWin32Error();
+                if (error == Interop.Errors.ERROR_INVALID_FUNCTION)
+                {
+                    // This is the case for Docker in CI.
+                    throw new SkipTestException($"GenerateConsoleCtrlEvent failed with ERROR_INVALID_FUNCTION. The process is not a console process or does not have a console.");
```

Comment:
OK, looks like it was that, after https://github.com/dotnet/runtime/pull/117105/commits/4af7d7df9d521fc33537c1e96311bbc4c5ad0d18, I can see the same error I see locally on docker. I will just skip the test in such case.  

I don't see a way to detect we are inside a docker container (in `PlatformDetection`) to make this check narrower.  I know we already don't fail on the [non-Docker windows case](https://dev.azure.com/dnceng-public/public/_build/results?buildId=1089316&view=logs&j=f967148e-4c68-5d25-c313-1c7f2944d30b&t=13d08b9b-6cf2-52cd-110c-5da23fc44a7b).